### PR TITLE
Renamed _symbol_ methods to _variable_ methods.

### PR DESCRIPTION
**Pull request by MichaelClerx**
## Description
Renamed _symbol_ methods to _variable_ methods.

## Motivation and Context
#230 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have updated all documentation necessary.
- [x] I have checked spelling in (new) comments.

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Also changed `get_derivative_symbols` to `get_derivatives` (as a derivative is not a sympy symbol)

**Comment by MichaelClerx** [comment]
Windows & coverage testing failing due to #233

**Comment by MichaelClerx** [comment]
Hi all!

Merging this will break FC and codegen (but easily fixed!).

When would be a good time?

**Comment by MauriceHendrix** [comment]
> Hi all!
> 
> Merging this will break FC and codegen (but easily fixed!).
> 
> When would be a good time?

now is as good (or bad) a time as any for me.

File: cellmlmanip/model.py [hunk 1]
```diff
@@ -745,9 +743,9 @@ def remove_variable(self, symbol):
                 self.rdf.remove(triple)
 
         # Remove references to variable and invalidate cache
-        del self._name_to_symbol[symbol.name]
+        del self._name_to_variable[symbol.name]
```
Comment (jonc125) [inline cellmlmanip/model.py]: Should we also change local variable names like here `symbol` becomes `variable`?

Comment (MichaelClerx) [reply cellmlmanip/model.py]: Done!

**Comment by MauriceHendrix** [comment]
Just added a fix (and test) to allow conversion of variables without cmeta_id, as in chaste_codegen some models I'm converting the free variable (time into milliseconds) without it having a cmeta_id

**Comment by MichaelClerx** [comment]
OK, ready to merge I think (once the online tests pass):

- Chaste codegen PR: Not public, I think, but @MauriceHendrix says he's ready
- FC PR: https://github.com/ModellingWebLab/weblab-fc/pull/115

File: cellmlmanip/model.py [hunk 2]
```diff
@@ -229,24 +230,26 @@ def transform_constants(self):
...
-        """Given the source and target component and variable, create a connection by assigning
-        the symbol from the source to the target. If units are not the same, it will add an equation
-        to the target component reflecting the relationship. If a symbol has not been assigned to
-        the source variable, then return False.
+        """Tells this model that the variable indicated by ``target_name`` should get its value from the variable
+        indicated by ``source_name``.
 
-        :param source_name: source variable name
-        :param target_name: target variable name
+        If the variables' units are not the same, and equation will be added to the target component.
```
Comment (jonc125) [inline cellmlmanip/model.py]: s/and/an/

File: cellmlmanip/model.py [hunk 3]
```diff
@@ -229,24 +230,26 @@ def transform_constants(self):
...
+
+        This method will only work if the source variable has been assigned a value, either through an equation or by
+        connecting it to a variable with an equation. If this is not yet the case, ``False`` is returned. If
```
Comment (jonc125) [inline cellmlmanip/model.py]: "connecting it" -> "connecting it (directly or indirectly)" ?

**Pull request merged**